packages/graalvm/src/main/kotlin/elide/runtime/gvm/internals/node/buffer/NodeBuffer.kt

```diff
@@ -12,3 +12,299 @@
  */
 package elide.runtime.gvm.internals.node.buffer
 
+import org.graalvm.polyglot.Value
+import org.graalvm.polyglot.proxy.ProxyInstantiable
+import java.nio.ByteBuffer
+import elide.annotations.Inject
+import elide.annotations.Singleton
+import elide.runtime.gvm.internals.intrinsics.Intrinsic
+import elide.runtime.gvm.internals.intrinsics.js.AbstractJsIntrinsic
+import elide.runtime.gvm.internals.intrinsics.js.AbstractNodeBuiltinModule
+import elide.runtime.gvm.internals.intrinsics.js.JsSymbol.JsSymbols.asJsSymbol
+import elide.runtime.intrinsics.GuestIntrinsic.MutableIntrinsicBindings
+import elide.runtime.intrinsics.js.node.BufferAPI
+import elide.runtime.intrinsics.js.node.buffer.Buffer
+
+// Symbol at which the main `Buffer` intrinsic is installed.
+private const val BUFFER_SYMBOL = "Buffer"
+
+// Internal symbol where the Node built-in module is installed.
+private const val BUFFER_MODULE_SYMBOL = "__Elide_node_buffer__"
+
+// Default setting for direct buffers.
+private const val DEFAULT_DIRECT_BUFFERS = false
+
+/**
+ * TBD.
+ */
+internal interface ByteAccessor {
+  /**
+   * TBD.
+   */
+  val mutable: Boolean get() = false
```

Comment:
I'll need to finish implementing this to unblock `fs`, which I plan to land in `alpha9`